### PR TITLE
OSDOCS#4612: Document the understanding Network Observability page

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.12

Issue: https://issues.redhat.com/browse/OSDOCS-4612

Link to docs preview: https://53508--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/understanding-network-observability-operator.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: networking/network_observability/understanding-network-observability-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+:_content-type: ASSEMBLY
+[id="network-observability-operator"]
+= Network Observability Operator in {product-title}
+include::_attributes/common-attributes.adoc[]
+:context: network-observability-operator
+
+toc::[]
+
+Network observability is an OpenShift operator that deploys a monitoring pipeline to collect and enrich network flows that are produced by the Network observability eBPF agent.
```
Comment (memodi) [inline networking/network_observability/understanding-network-observability-operator.adoc:9]:
```suggestion
Network observability is an OpenShift operator that deploys a monitoring pipeline to collect and enrich network traffic flows that are produced by the Network observability eBPF agent.
```

**Review by memodi — COMMENTED** [review]
adding comments/suggestions.

File: networking/network_observability/understanding-network-observability-operator.adoc (hunk 1 above)
Comment (xenolinux) [reply networking/network_observability/understanding-network-observability-operator.adoc]: Addressed

File: modules/nw-view-status-configuration-network-observability-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/ Module included in the following assemblies:
+// * networking/network_observability/understanding-network-observability-operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-status-configuration-network-observability-operator_{context}"]
+= Viewing Network Observability Operator status and configuration
+
+You can inspect the status and view the details of the Network Observability Operator using the `oc describe` command.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-view-status-configuration-network-observability-operator.adoc:8]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+$ oc get -n netobserv deployment/network-observability
+----
++
+.Example output
+----
+NAME                    READY   UP-TO-DATE   AVAILABLE   AGE
+network-observability   1/1     1            1           56m
+----
+
+. Run the following command to view the state of the Network Observability  Operator:
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:27]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+.Procedure
+
+The Network Observability Operator is deployed during installation as a Kubernetes `Deployment`.
+
+. Run the following command to view the Deployment status:
++
+[source,terminal]
+----
+$ oc get -n netobserv deployment/network-observability
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:18]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+flowlogs-pipeline-56hbp           1/1     Running   0          147m
+flowlogs-pipeline-9plvv           1/1     Running   0          147m
+flowlogs-pipeline-h5gkb           1/1     Running   0          147m
+flowlogs-pipeline-hh6kf           1/1     Running   0          147m
+flowlogs-pipeline-w7vv5           1/1     Running   0          147m
+loki                              1/1     Running   0          145m
+netobserv-plugin-cdd7dc6c-j8ggp   1/1     Running   0          147m
+----
+
+Flowlogs pipeline pods collect flows, enriches the collected flows, then send flows to the Loki storage.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 6]
```diff
@@ -0,0 +1,79 @@
+/ Module included in the following assemblies:
+// * networking/network_observability/understanding-network-observability-operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-network-observability-operator_{context}"]
+= Network Observability Operator
+
+When you deploy flow collector is deployed, the Operator provides dashboards, metrics, and keeps flows accessible in a queryable log store, Grafana Loki. When a FlowCollector is deployed, new dashboards are available in the Console.
```
Comment (davegord) [inline modules/nw-network-observability-operator.adoc:8]: This hurts my head.  What are we trying to convey here? 
I'm not sure but I think we are trying to talk about the flow collector?
I don't know why we would say anything about a queryable log store.  
I also don't know why we would talk about deploying flowcollector and a new dashboard appears.

I think we want to talk about the overall network observability installation - and part of that is deploying a flow collector.  Upon successful install and configuration of the flow collector, a new network section is available in the ocp console's observe section.

**Review by stleerh — COMMENTED** [review]
I'm not clear where this belongs in the Network Observability document.  The outline didn't have this section.

File: modules/nw-network-observability-operator.adoc [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+[source,terminal]
+----
+$ oc get -n netobserv deployment/netobserv-controller-manager
+----
++
+.Example output
+----
+NAME                    READY   UP-TO-DATE   AVAILABLE   AGE
+network-observability   1/1     1            1           56m
+----
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc]: Addressed- Removed

**Review by jotak — APPROVED** [review]
This looks good to me, thanks @xenolinux

File: modules/nw-network-observability-operator.adoc [hunk 8]
```diff
@@ -0,0 +1,64 @@
+/ Module included in the following assemblies:
+// * networking/network_observability/understanding-network-observability-operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-network-observability-operator_{context}"]
+= Network Observability Operator
+
+When you deploy flow collector, the Operator provides dashboards, metrics, and keeps flows accessible in a queryable log store, Grafana Loki. When a FlowCollector is deployed, new dashboards are available in the Console.
```
Comment (nathan-weinberg) [inline modules/nw-network-observability-operator.adoc:8]:
```suggestion
When you deploy flow collector, the Operator provides dashboards, metrics, and keeps flows accessible in a queryable log store, Grafana Loki. When a FlowCollector is deployed, new dashboards are available in the Console.
```
We should probably standardize `flow collector` vs `FlowCollector`? Not sure which is more appropriate

File: modules/nw-network-observability-operator.adoc [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
++
+.Example output
+----
+NAME                              READY   STATUS    RESTARTS   AGE
+flowlogs-pipeline-56hbp           1/1     Running   0          147m
+flowlogs-pipeline-9plvv           1/1     Running   0          147m
+flowlogs-pipeline-h5gkb           1/1     Running   0          147m
+flowlogs-pipeline-hh6kf           1/1     Running   0          147m
+flowlogs-pipeline-w7vv5           1/1     Running   0          147m
+loki                              1/1     Running   0          145m
```
Comment (nathan-weinberg) [inline modules/nw-network-observability-operator.adoc:40]: Just a small note, if Loki Operator is used those Loki pods will be under the `openshift-operators-redhat` NS, example below
```bash
[nathan@nathan-redhat ocp]$ oc get pods -n openshift-operators-redhat
NAME                                                READY   STATUS    RESTARTS   AGE
loki-operator-controller-manager-5f6cff4f9d-jq25h   2/2     Running   0          18h
lokistack-compactor-0                               1/1     Running   0          18h
lokistack-distributor-654f87c5bc-qhkhv              1/1     Running   0          18h
lokistack-distributor-654f87c5bc-skxgm              1/1     Running   0          18h
lokistack-gateway-796dc6ff7-c54gz                   2/2     Running   0          18h
lokistack-index-gateway-0                           1/1     Running   0          18h
lokistack-index-gateway-1                           1/1     Running   0          18h
lokistack-ingester-0                                1/1     Running   0          18h
lokistack-ingester-1                                1/1     Running   0          18h
lokistack-ingester-2                                1/1     Running   0          18h
lokistack-querier-66747dc666-6vh5x                  1/1     Running   0          18h
lokistack-querier-66747dc666-cjr45                  1/1     Running   0          18h
lokistack-querier-66747dc666-xh8rq                  1/1     Running   0          18h
lokistack-query-frontend-85c6db4fbd-b2xfb           1/1     Running   0          18h
lokistack-query-frontend-85c6db4fbd-jm94f           1/1     Running   0          18h
```

File: networking/network_observability/understanding-network-observability-operator.adoc [hunk 10]
```diff
@@ -0,0 +1,13 @@
+:_content-type: ASSEMBLY
+[id="network-observability-operator"]
+= Network Observability Operator in {product-title}
+include::_attributes/common-attributes.adoc[]
+:context: network-observability-operator
+
+toc::[]
+
+Network observability is an OpenShift operator that deploys a monitoring pipeline to collect and enrich network traffic flows that are produced by the Network observability eBPF agent.
```
Comment (nathan-weinberg) [inline networking/network_observability/understanding-network-observability-operator.adoc:9]: Should we capitalize `observability` here?

Comment (xenolinux) [reply networking/network_observability/understanding-network-observability-operator.adoc:9]: Yes good catch - updated

File: modules/nw-network-observability-operator.adoc (hunk 9 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:40]: Addressed

File: networking/network_observability/understanding-network-observability-operator.adoc (hunk 10 above)
Comment (xenolinux) [reply networking/network_observability/understanding-network-observability-operator.adoc]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 11]
```diff
@@ -0,0 +1,91 @@
+/ Module included in the following assemblies:
+// * networking/network_observability/understanding-network-observability-operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-network-observability-operator_{context}"]
+= Network Observability Operator
+
+When you deploy flow collector, the Operator provides dashboards, metrics, and keeps flows accessible in a queryable log store, Grafana Loki. When a flow collector is deployed, new dashboards are available in the Console.
```
Comment (nathan-weinberg) [inline modules/nw-network-observability-operator.adoc:8]: Is it always in Grafana Loki? cc @jotak @mariomac @jpinsonneau

Comment (jotak) [reply modules/nw-network-observability-operator.adoc:8]: Yes there is no other option so far

**Review by mariomac — APPROVED** [review]

File: modules/nw-network-observability-operator.adoc (hunk 9 above)
Comment (memodi) [reply modules/nw-network-observability-operator.adoc:40]: we should remove loki pod as output here, since IMO we're only showing NOO workloads here.

Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 12]
```diff
@@ -0,0 +1,90 @@
+/ Module included in the following assemblies:
```
Comment (memodi) [inline modules/nw-network-observability-operator.adoc:1]: this line is showing in preview, is it supposed to?

File: modules/nw-view-status-configuration-network-observability-operator.adoc [hunk 13]
```diff
@@ -0,0 +1,17 @@
+/ Module included in the following assemblies:
```
Comment (memodi) [inline modules/nw-view-status-configuration-network-observability-operator.adoc:1]: same here, not sure if it should be visible in the docs?

File: modules/nw-network-observability-operator.adoc (hunk 12 above)
Comment (memodi) [inline modules/nw-network-observability-operator.adoc:1]:
```suggestion
// Module included in the following assemblies:
```

File: modules/nw-view-status-configuration-network-observability-operator.adoc (hunk 13 above)
Comment (memodi) [inline modules/nw-view-status-configuration-network-observability-operator.adoc:1]:
```suggestion
// Module included in the following assemblies:
```

File: modules/nw-network-observability-operator.adoc (hunk 12 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:1]: hmm. good catch. Thanks for noticing

File: modules/nw-view-status-configuration-network-observability-operator.adoc (hunk 13 above)
Comment (xenolinux) [reply modules/nw-view-status-configuration-network-observability-operator.adoc]: Addressed

File: modules/nw-network-observability-operator.adoc (hunk 12 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc]: Addressed

File: modules/nw-view-status-configuration-network-observability-operator.adoc (hunk 13 above)
Comment (xenolinux) [reply modules/nw-view-status-configuration-network-observability-operator.adoc]: Addressed

File: networking/network_observability/understanding-network-observability-operator.adoc [hunk 14]
```diff
@@ -0,0 +1,13 @@
+:_content-type: ASSEMBLY
+[id="nw-network-observability-operator"]
+= Network Observability Operator in {product-title}
+include::_attributes/common-attributes.adoc[]
+:context: nw-network-observability-operator
+
+toc::[]
+
+Network observability is an OpenShift operator that deploys a monitoring pipeline to collect and enrich network traffic flows that are produced by the Network Observability eBPF agent.
```
Comment (nathan-weinberg) [inline networking/network_observability/understanding-network-observability-operator.adoc:9]:
```suggestion
Network Observability is an OpenShift operator that deploys a monitoring pipeline to collect and enrich network traffic flows that are produced by the Network Observability eBPF agent.
```

**Review by jpinsonneau — APPROVED** [review]

File: networking/network_observability/understanding-network-observability-operator.adoc (hunk 14 above)
Comment (xenolinux) [reply networking/network_observability/understanding-network-observability-operator.adoc]: Addressed

File: modules/nw-network-observability-operator.adoc [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+$ oc get flowcollector/cluster
+----
++
+.Example output
+----
+NAME      AGENT   SAMPLING (EBPF)   DEPLOYMENT MODEL   STATUS
+cluster   EBPF    50                DIRECT             Ready
+----
+
+. Check the status of pods running in namespace `netobserv`.
```
Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:25]: Maybe...

. Check the status of pods running in the `netobserv` namespace by entering the following command:

File: modules/nw-network-observability-operator.adoc [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+NAME                              READY   STATUS    RESTARTS   AGE
+flowlogs-pipeline-56hbp           1/1     Running   0          147m
+flowlogs-pipeline-9plvv           1/1     Running   0          147m
+flowlogs-pipeline-h5gkb           1/1     Running   0          147m
+flowlogs-pipeline-hh6kf           1/1     Running   0          147m
+flowlogs-pipeline-w7vv5           1/1     Running   0          147m
+netobserv-plugin-cdd7dc6c-j8ggp   1/1     Running   0          147m
+----
+
+Flowlogs-pipeline pods collect flows, enriches the collected flows, then send flows to the Loki storage.
```
Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:43]: I'm conflicted on how to approach/change this. Generally, we only do 1 procedure per module, and this reads as 3 different procedures. Should these be three separate modules?

Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:43]: Should it be `flowlogs-pipeline` in backticks? I'm thinking yes, but could be wrong.

Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:43]: Should it be `Netobserv` plugin pods? I'm thinking yes.

File: modules/nw-network-observability-operator.adoc [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+Netobserv plugin pods create a visualization plugin for the {product-title} Console.
+
+. Check the status of pods running in namespace `netobserv-privileged`.
```
Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:46]: . Check the status of pods running in the `netobserv-privileged` 
 namespace by running the following command:

File: modules/nw-network-observability-operator.adoc [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+----
+NAME                         READY   STATUS    RESTARTS   AGE
+netobserv-ebpf-agent-4lpp6   1/1     Running   0          151m
+netobserv-ebpf-agent-6gbrk   1/1     Running   0          151m
+netobserv-ebpf-agent-klpl9   1/1     Running   0          151m
+netobserv-ebpf-agent-vrcnf   1/1     Running   0          151m
+netobserv-ebpf-agent-xf5jh   1/1     Running   0          151m
+----
+
+Netobserv eBPF agent pods monitor the nodes network interfaces to get flows and send them to Flowlogs Pipeline pods.
```
Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:63]: Suggestion: 

s/ nodes / node's

Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:63]: Above it is written as "Flowlog-pipeline" (with a dash). 

Should it be written with a dash here? And should we put it in backticks?

File: modules/nw-network-observability-operator.adoc [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+
+. If you are using a Loki Operator, check the status of pods running in namespace `openshift-operators-redhat`.
```
Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:65]: . If you are using a Loki Operator, check the status of pods running in the `openshift-operators-redhat`namespace by entering the following command:

File: modules/nw-network-observability-operator.adoc (hunk 18 above)
Comment (stevsmit) [inline modules/nw-network-observability-operator.adoc:63]: If we decide Netobserv is in backticks, it should be in backticks here as well.

**Comment by stevsmit** [comment]
@xenolinux a few nits. Looks good overall.

File: modules/nw-network-observability-operator.adoc [hunk 20]
```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+// * networking/network_observability/understanding-network-observability-operator.adoc
+
+:_content-type: PROCEDURE
+[id="nw-network-observability-operator_{context}"]
+= Network Observability Operator
+
+When you deploy flow collector, the Operator provides dashboards, metrics, and keeps flows accessible in a queryable log store, Grafana Loki. When a flow collector is deployed, new dashboards are available in the Console.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:8]: Addressed

[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:8]: Addressed

File: modules/nw-network-observability-operator.adoc (hunk 15 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:25]: Addressed

File: modules/nw-network-observability-operator.adoc (hunk 16 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:43]: yes that's right- addressed

Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:43]: it should be `netobserv-plugin` pods - addressed

File: modules/nw-network-observability-operator.adoc (hunk 17 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:46]: Addressed

File: modules/nw-network-observability-operator.adoc (hunk 16 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:43]: This should be one procedure only.

File: modules/nw-network-observability-operator.adoc (hunk 18 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:63]: Addressed

Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:63]: yes. Addressed

File: modules/nw-network-observability-operator.adoc (hunk 19 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:65]: addressed

File: modules/nw-network-observability-operator.adoc (hunk 18 above)
Comment (xenolinux) [reply modules/nw-network-observability-operator.adoc:63]: Addressed

File: _topic_maps/_topic_map.yml [hunk 21]
```diff
@@ -1317,6 +1317,8 @@ Topics:
     File: installing-operators
   - Name: Observing the network traffic
     File: observing-network-traffic
+  - Name: Understanding Network Observability Operator
```
Comment (skrthomas) [inline _topic_maps/_topic_map.yml:1320]: @xenolinux do you think we should move this in the topic map under the Overview instead of under Observing?

**Pull request merged**